### ISSUE TITLE
[Task 24] Reduce Colocate Phase-Switch Overhead with Bidirectional GPU Weight Handoff

### ISSUE THREAD

**Issue by fxfxfxfxfxfxfxfx** [redai-infra/Relax#169]
## Background and Motivation

In the four-GPU Qwen3-VL-4B colocate baseline, rollout and training share the same GPUs. Therefore, phase-switch overhead directly blocks end-to-end training.

| Baseline metric | Time | Percentage of step time |
|---|---:|---:|
| Mean step time | 22.361 s | 100% |
| Rollout -> Train | 2.736 s | 12.2% |
| Train -> Rollout | 10.530 s | 47.1% |
| Total transition proxy | 13.266 s | 59.3% |

The legacy transition proxy is `sleep + wake_up + update_weights`. These values are host-observed wall times, not GPU kernel-active times.

Approximately 59% of the baseline step time is spent switching phases. The existing path also stages actor weights through Host memory and repeatedly releases and restores phase-specific states.

This task proposes bidirectional GPU weight handoff, selective non-weight memory release, and overlap with optimizer offload to reduce this blocking overhead.

## Repository and Development Branch

- Repository: `redai-infra/Relax`
- Upstream base commit: `cfda05351d4b137f60e2b71d048beab1316d9d9f`
- Fork: `fxfxfxfxfxfxfxfx/Relax`
- Development branch: `task24-colocate-weight-handoff`
- Current development commit: `6e55ad0f2fd854a55d015fd06e12d76fec3dc346`
- The A/B benchmark uses the same implementation commit and differs only in `--colocate-weight-handoff`.

## Environment

- Base image: `lmsysorg/sglang:v0.5.12.post1-cu129`
- OS: Ubuntu 22.04.5
- Python: 3.12.3
- CUDA: 12.9
- NVIDIA driver: 580.105.08
- PyTorch: 2.11.0+cu129
- Ray: 2.56.1
- Transformers: 5.6.0
- SGLang: 0.5.12.post1
- SGLang base commit: `5a15cde858ea09b77116212a39356f2fc51b8584`
- Megatron-Bridge commit: `2faedbf6fe3c422835a44b2b360cadcb2a116a54`
- Megatron-LM commit: `85bced0ae6ab46f61a0fd774074a3273daf6ae02`
- GPU: 4 x NVIDIA A800 80GB PCIe
- CPU: Intel Xeon Gold 6348, 112 logical CPUs
- System memory: 1 TiB, no swap

## Entry Script and Commands

Entry script:

```text
scripts/training/multimodal/run-qwen3-vl-4B-8xgpu.sh
```

### Common Configuration

```bash
export CUDA_VISIBLE_DEVICES=0,1,2,3
export NUM_GPUS=4
export TASK_GPUS=4
